File: snippets/lesson-6-subject.cs
using System;
using System.Reactive.Disposables;
using System.Reactive.Subjects;

var mySubject = new Subject<int>();
var disposables = new CompositeDisposable();

disposables.Add(mySubject.Subscribe(v => Console.WriteLine($"Sub 1: {v}")));
disposables.Add(mySubject.Subscribe(v => Console.WriteLine($"Sub 2: {v}")));

mySubject.OnNext(100);

disposables.Dispose();
Console.WriteLine("(subscriptions disposed)");

File: snippets/lesson-6-subject.ts
import { Subject, Subscription } from 'rxjs';

const mySubject = new Subject<number>();

const sub1: Subscription = mySubject.subscribe((v: number) => console.log(`Sub 1: ${v}`));
const sub2: Subscription = mySubject.subscribe((v: number) => console.log(`Sub 2: ${v}`));

mySubject.next(100);

sub1.unsubscribe();
sub2.unsubscribe();
console.log('(subscriptions disposed)');